vividus-extension-selenium/src/main/java/org/vividus/ui/listener/SourceCodePublishingOnFailureListener.java

```diff
@@ -48,6 +49,10 @@ public void onAssertionFailure(AssertionFailedEvent event)
         if (publishSourceOnFailure && webDriverProvider.isWebDriverInitialized())
         {
             contextSourceCodeProvider.getSourceCode().forEach(this::publishSource);
+            if (publishShadowDomSourceOnFailure)
+            {
+                publishShadowDomSource();
```

Comment:
Generic `SourceCodePublishingOnFailureListener` should know nothing about Shadow DOM, because this class is used in both web-app and mobile-app plugins